lighthouse-core/audits/dobetterweb/link-blocking-first-paint.js

```diff
@@ -55,31 +56,42 @@ class LinkBlockingFirstPaintAudit extends Audit {
       };
     }
 
-    let totalSpendTime = 0;
-    const results = artifact.items
-      .filter(item => item.tag.tagName === tagFilter)
-      .map(item => {
-        totalSpendTime += item.spendTime;
+    const filtered = artifact.filter(item => item.tag.tagName === tagFilter);
 
-        return {
-          url: item.tag.url,
-          label: `delayed first paint by ${item.spendTime}ms`
-        };
-      });
+    const startTime = filtered.reduce((t, item) => Math.min(t, item.startTime), Number.MAX_VALUE);
+    let endTime = 0;
 
+    const results = filtered.map(item => {
+      endTime = Math.max(item.endTime, endTime);
+
+      return {
+        url: URL.getDisplayName(item.tag.url),
+        totalKb: `${Math.round(item.transferSize / 1024)} KB`,
+        totalMs: `${Math.round((item.endTime - startTime) * 1000)}ms`
```

Comment:
> Using the overall start then highlights the biggest offenders the right way, since these block first paint, we want developers to target the ones that pushed it out the farthest regardless of when the request was started by Chrome

Ah, I see now. So the idea is that the parser has seen these by the time the first request is sent, so even if it doesn't get around to requesting this item right away, it makes sense to punish for that full time? That makes sense for many cases it's of course not just that item's contribution to blocking time...e.g. if the browser is throttled on the number of connections to make and so a request gets pushed further out, eliminating that particular request won't speed up page by `item.endTime - startTime`.

OTOH, maybe with the change you're making to how the total blocked time is calculated, it's clear that while this individual request blocked first paint for this long, eliminating this request won't speed up first paint by that much (i.e. the individual delays are overlapping).

I might be out in the weeds here, though, so I'll defer to you all :) @paulirish for his opinion